### ISSUE TITLE
Handle numba TypingError

### ISSUE THREAD

**Issue by Vayel** [jmcarpenter2/swifter#35]
```python
import numba
import pandas as pd
import swifter


@numba.jit(nopython=True)
def model(x, y):
    return x + y


def model_wrapper(parameters):
    return model(**parameters)


if __name__ == "__main__":
    df = pd.DataFrame({'x': [1, 2, 3, 4], 'y': [5, 6, 7, 8]})
    df.swifter.apply(model_wrapper, axis=1)
```

When `swifter` tries to vectorize the function, I get:

```
Traceback (most recent call last):
  File "test_swifter.py", line 17, in <module>
    df.swifter.apply(model_wrapper, axis=1)
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/swifter/swifter.py", line 236, in apply
    tmp_df = func(samp, *args, **kwds)
  File "test_swifter.py", line 12, in model_wrapper
    return model(**parameters)
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/dispatcher.py", line 348, in _compile_for_args
    error_rewrite(e, 'typing')
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/dispatcher.py", line 315, in error_rewrite
    reraise(type(e), e, None)
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/six.py", line 658, in reraise
    raise value.with_traceback(tb)
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Internal error at <numba.typeinfer.ArgConstraint object at 0x7fbda191d160>:
--%<----------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/errors.py", line 599, in new_error_context
    yield
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/typeinfer.py", line 199, in __call__
    assert ty.is_precise()
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/typeinfer.py", line 142, in propagate
    constraint(typeinfer)
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/typeinfer.py", line 200, in __call__
    typeinfer.add_type(self.dst, ty, loc=self.loc)
  File "/home/simudyne/anaconda3/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/errors.py", line 607, in new_error_context
    six.reraise(type(newerr), newerr, tb)
  File "/home/simudyne/.local/share/virtualenvs/lebaron-models-liy-Dh9K/lib/python3.7/site-packages/numba/six.py", line 659, in reraise
    raise value
numba.errors.InternalError: 
[1] During: typing of argument at test_swifter.py (8)
--%<----------------------------------------------------------------------------


File "test_swifter.py", line 8:
def model(x, y):
    return x + y
    ^

This error may have been caused by the following argument(s):
- argument 0: cannot determine Numba type of <class 'pandas.core.series.Series'>
- argument 1: cannot determine Numba type of <class 'pandas.core.series.Series'>

This is not usually a problem with Numba itself but instead often caused by
the use of unsupported features or an issue in resolving types.

To see Python/NumPy features supported by the latest release of Numba visit:
http://numba.pydata.org/numba-doc/dev/reference/pysupported.html
and
http://numba.pydata.org/numba-doc/dev/reference/numpysupported.html

For more information about typing errors and how to debug them visit:
http://numba.pydata.org/numba-doc/latest/user/troubleshoot.html#my-code-doesn-t-compile

If you think your code should work with Numba, please report the error message
and traceback, along with a minimal reproducer at:
https://github.com/numba/numba/issues/new
```

This is because `numba.errors.TypingError` is not caught in [swifter.py#L251](https://github.com/jmcarpenter2/swifter/blob/master/swifter/swifter.py#L251).

To fix this, I do:

```python
def model_wrapper(parameters):
    if not isinstance(parameters, pd.Series):
        raise TypeError()
    return model(**parameters)
```

It would be nice if `swifter` could handle this itself (or, at least, add a note in the documentation).

Thank you!

**Comment by jmcarpenter2** [comment]
Hi @Vayel,

This is now fixed in version 0.273. Please upgrade using either
`pip install -U swifter`
or
`pip install swifter==0.273`

Thanks for so clearly laying out the issue. This was an easy fix and I apologize for taking so long to do it.

Your contribution is appreciated!
Jason

**Comment by Vayel** [comment]
Thank you so much @jmcarpenter2!
